blockchain/api.go

```diff
@@ -1364,24 +1369,43 @@ func (api *BasicEventsAPI) GetLastBlock(ctx context.Context) (uint64, error) {
 
 func (api *BasicEventsAPI) GetEvents(ctx context.Context, filter *EventFilter) (chan *Event, error) {
 	var out = make(chan *Event, 128)
+	// Number of blocks already present in blockchain that we will read in batches.
+	lastBlock, err := api.GetLastBlock(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest block number: %v", err)
+	}
+
+	if filter.FromBlock.Uint64() > lastBlock {
```

Comment:
why not just wait?

Reply:
because it's weird to wait for _something in the future_, not going to change